pkg/api/types.go

```diff
@@ -121,3 +121,26 @@ type InstallResult struct {
 	// Error describes last error encountered during install operation
 	Error error
 }
+
+// SourceInfo stores information about the source code
+type SourceInfo struct {
+	// Store the current GIT ref which might be a branch or commit id
+	// This information is available in the image as 'io.openshift.build.commit.ref' LABEL
+	Ref string
+
+	// The GIT commit id
+	// This information is available in the image as 'io.openshift.build.commit.id' LABEL
+	ID string
```

Comment:
behavior for these fieids:

ref is a commit.  commitid will match ref
ref is a branch.  commitid will be a commit in that branch
ref is a tag.  commitid will be the commit pointed to by the tag.